src/dmd/parse.d

```diff
@@ -4493,7 +4482,7 @@ final class Parser(AST) : Lexer
                         if (udas)
                             error("user-defined attributes not allowed for `%s` declarations", Token.toChars(tok));
 
-                        t = parseType();
+                        AST.Type t = parseType();
```

Comment:
`auto`

Reply:
Is there a general guideline for auto use?  Is it always preferred when possible?

Reply:
Yes, or even better `const`/`immutable`, where possible. But it should usually be avoided for the return type of functions, unless it's a complicated template type.